api/v1beta1/hetznercluster_webhook.go

```diff
@@ -73,7 +73,12 @@ func (r *HetznerCluster) ValidateCreate() (admission.Warnings, error) {
 	hetznerclusterlog.V(1).Info("validate create", "name", r.Name)
 	var allErrs field.ErrorList
 
-	if len(r.Spec.ControlPlaneRegions) == 0 {
+	allowEmptyControlPlaneAddress := "false"
```

Comment:
better: make allowEmptyControlPlaneAddress a boolean and set it on true if the annotation is set